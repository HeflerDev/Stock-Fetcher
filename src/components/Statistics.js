import React from 'react';
import PropTypes from 'prop-types';

const Statistics = ({ statistics }) => {
  const {
    ticker,
    changes,
    price,
    changesPercentage,
    companyName,
  } = statistics;

  const checkStatus = () => {
    if (changes >= 0) {
      return 'green';
    }
    return 'red';
  };

  return (
    <div className="board statistics">
      <div className="queue between stock-header">
        <div>{ ticker }</div>
        <div>{ companyName }</div>
      </div>
      <div className="queue center">
        <p>
          Price:
          { price }
        </p>
      </div>
      <div className="queue around">
        <p className={checkStatus()}>
          Changes:
          { changes }
        </p>
        <p className={checkStatus()}>
          Percentage %:
          { changesPercentage }
        </p>
      </div>
    </div>
  );
};

Statistics.propTypes = {
  statistics: PropTypes.shape({
    ticker: PropTypes.string.isRequired,
    changes: PropTypes.number.isRequired,
    price: PropTypes.string.isRequired,
    changesPercentage: PropTypes.string.isRequired,
    companyName: PropTypes.string.isRequired,
  }).isRequired,
};

export default Statistics;
