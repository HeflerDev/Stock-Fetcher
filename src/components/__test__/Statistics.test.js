import renderer from 'react-test-renderer';
import Statistics from '../Statistics';

describe('when loading statistics', () => {
  const mockProp = {
    ticker: '0',
    changes: 0,
    price: '1.0',
    changesPercentage: '%',
    companyName: 'Test',
  };

  it('render statistic correctly', () => {
    const tree = renderer
      .create(<Statistics statistics={mockProp} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
