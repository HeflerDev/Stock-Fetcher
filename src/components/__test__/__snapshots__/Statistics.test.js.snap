// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when loading statistics render statistic correctly 1`] = `
<div
  className="board statistics"
>
  <div
    className="queue between stock-header"
  >
    <div>
      0
    </div>
    <div>
      Test
    </div>
  </div>
  <div
    className="queue center"
  >
    <p>
      Price:
      1.0
    </p>
  </div>
  <div
    className="queue around"
  >
    <p
      className="green"
    >
      Changes:
      0
    </p>
    <p
      className="green"
    >
      Percentage %:
      %
    </p>
  </div>
</div>
`;
